molo/commenting/admin.py

```diff
@@ -1,8 +1,180 @@
 from django.contrib import admin
 
 from molo.commenting.models import MoloComment
+from molo.core.models import ArticlePage
 from django_comments.models import CommentFlag
+from django_comments.admin import CommentsAdmin
+from django.core.urlresolvers import reverse
+from django.core.exceptions import PermissionDenied
+from django.contrib.admin.views.main import ChangeList
+from django.shortcuts import get_object_or_404
+from django.contrib.admin.utils import unquote
+from django.contrib.contenttypes.models import ContentType
 
 
-admin.site.register(MoloComment)
+class MoloCommentAdmin(CommentsAdmin):
+    list_display = (
+        'comment', 'content', '_user', 'is_removed', 'is_reported',
+        'submit_date')
+
+    def is_reported(self, obj):
+        if (obj.flags.count() > 0):
+            return True
+        return False
+    is_reported.admin_order_field = 'is_reported'
+    is_reported.boolean = True
+
+    def get_user_display_name(self, obj):
+        if obj.name.lower().startswith('anon'):
+            return obj.user.username
+        return obj.name
+
+    def _user(self, obj):
+        url = reverse('admin:auth_user_change', args=(obj.user.id,))
+        return '<a href="?user=%s">%s</a>' % (
+            obj.user.id,
+            self.get_user_display_name(obj)
+        ) + ' (<a href="%s">edit</a>)' % url
+    _user.allow_tags = True
+    _user.short_description = 'User'
+
+    def content(self, obj, *args, **kwargs):
+        content_type = obj.content_type
+        if not int(obj.object_pk) in self.ct_map[content_type]:
+            content = obj
+        else:
+            content = self.ct_map[content_type][int(obj.object_pk)]
+        url = reverse('admin:%s_%s_moderate' % (
+            content_type.app_label,
+            content_type.model
+        ), args=(content.id,))
+
+        return '<a href="%s">%s</a>' % (url, content)
+
+    content.allow_tags = True
+    content.short_description = 'Content'
+
+    def get_changelist(self, request):
```

Comment:
I spent a long time trying to figure this out as well but best I can figure is it's to override the changelist result set so that we could limit it by article. I'm still not convinced there isn't a better way but I haven't been able to find it so far.
Maybe I can revisit it when I have more django experience under my belt :)